classes/blocks/class-articles.php

```diff
@@ -381,8 +380,21 @@ private function filter_posts_by_page_types_or_tags( &$fields ) {
 		// Get user defined tags from backend.
 		$tags = $fields['tags'] ?? [];
 
+		// Validate tag ids.
+		if ( ! empty( $tags ) ) {
+			$validated_tags = [];
```

Comment:
Small suggestion: `array_filter` is a bit easier to read, and you would not need the empty check and inner assignments
```$tags = array_filter($tags, function($tag_id) { return get_tag( $tag_id ) instanceof \WP_Term; }```